provider/core/README.md

```diff
@@ -41,16 +41,9 @@ This crate also contains some concrete implementations for testing purposes:
 Types compatible with [`Yokeable`] can be passed through the data provider, so long as they are
 associated with a marker type implementing [`DataMarker`].
 
-Most [`DataProvider`] traits take two lifetime arguments: `'d` and `'s`. These represent data
-lifetimes for different ownership models of the data.
-
-- `'d` is the lifetime of a reference to the whole data struct being borrowed.
-- `'s` is the lifetime of fields within a data struct.
-
-Data structs and [`DataMarker`] use the `'s` lifetime, and this is also the lifetime that the
-[`Yokeable`] implementation should handle.
-
-`'s` should exceed `'d` (i.e., `'s: 'd`). If a data struct is fully owned, `'s: 'static`.
+Most [`DataProvider`] traits take one lifetime argument: `'data`. This represents the lifetime
+of fields within a data struct. [`DataMarker`] also takes the `'data` lifetime, and this is
```

Comment:
I think we should expand this to mention the cart, `'data` is the lifetime of data in the _cart_, which the data struct can borrow from.

Reply:
Will follow up with this and some more cleanups